### PR TITLE
Sultan 2.0

### PR DESCRIPTION
**Pull request by ahmedaljazzar**
This is a new release of Sultan that moves the operations from `make` to `shell`. It has almost the same functionality as the previous version with some new changes on the image workflow.

## What's new
- Makefiles are now deprecated, they are left in place as reference until we make sure the shell scripts are stable.
- VMs can be created from any image even if that image wasn't created by the same user.
- Images can be created with custom names. (You can create as many images as you want)
- Env files are now customized to be compatible with the shell scripts instead of the Makefiles.
- configuration files are moved to `configs/` directory from the root directory.
- Unified playbooks syntax.
- The devstack on the remote machine is using a Python 3 virtualenv instead of a Python 2.
- The VM stores an ENV variable of the last user who created the machine.

## How to start
- Make sure you update your configuration file.
```console
$ export $PATH=$PATH:/path/to/sultan
$ sultan config init
# Manually copy the old values from `.configs.$USER` into `configs/.configs.$USER`.
```
- OPTIONAL: Create an image for your current instance.
```
$ sultan image create
```
- Create a new Juniper devstack
```
$ sultan instance create --image juniper-devstack
```

## How you can help
- Reviewing the changes.
- Enhancing the documentation.
- Suggesting new features.

### CODE REVIEW

**Comment by thraxil** [comment]
My first attempt with a clean checkout and following the quickstart instructions in the README:

```
$ ./sultan config init
./sultan: 6: source: not found
./sultan: 7: source: not found
./sultan: 29: Syntax error: Bad function name
```

`/bin/sh` on Ubuntu is actually `dash`, which doesn't support `source`. Unless we are trying to support some ancient version of AIX or something, we are probably better off explicitly using `/bin/bash` for all the scripts.

Documentation-wise, I'd suggest continuing the quickstart all the way through to getting a running devstack. As it is, a new user following the instructions gets their config created, then the documentation veers off into discussions of various other possible setups and advanced options that they might want. They have to then read through all of that to eventually figure out that `sultan instance setup` comes next.

I haven't investigated why yet, but it's also a bit annoying that it doesn't seem to be able to pick up on my running `ssh-agent`. I get prompted for the passphrase for my ssh key about half a dozen times during the process.

File: scripts/configurations.sh [hunk 1]
```diff
@@ -0,0 +1,102 @@
+#!/bin/sh
```
Comment (thraxil) [inline scripts/configurations.sh:1]:
```suggestion
#!/bin/bash
```

File: scripts/devstack.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
+#!/bin/sh
```
Comment (thraxil) [inline scripts/devstack.sh:1]:
```suggestion
#!/bin/bash
```

File: scripts/firewall.sh [hunk 3]
```diff
@@ -0,0 +1,124 @@
+#!/bin/sh
```
Comment (thraxil) [inline scripts/firewall.sh:1]:
```suggestion
#!/bin/bash
```

File: scripts/image.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
```
Comment (thraxil) [inline scripts/image.sh:1]:
```suggestion
#!/bin/bash
```

File: scripts/instance.sh [hunk 5]
```diff
@@ -0,0 +1,226 @@
+#!/bin/sh
```
Comment (thraxil) [inline scripts/instance.sh:1]:
```suggestion
#!/bin/bash
```

File: scripts/local.sh [hunk 6]
```diff
@@ -0,0 +1,119 @@
+#!/bin/sh
```
Comment (thraxil) [inline scripts/local.sh:1]:
```suggestion
#!/bin/bash
```

**Review by thraxil — CHANGES_REQUESTED** [review]

**Comment by ahmedaljazzar** [comment]
Thanks @thraxil for the review. I just applied the changes you requested and tested them. I'll be looking into the ssh key issue you had.

**Review by thraxil — APPROVED** [review]

**Comment by ahmedaljazzar** [comment]
Merging and will continue updating in separate PRs

**Pull request merged**